client/router.js

```diff
@@ -369,3 +427,45 @@ Router.prototype._updateCallbacks = function () {
 
 Router.prototype._page = page;
 Router.prototype._qs = qs;
+
+function shouldCallTrigger(current, fn) {
+  var name = current.route.name;
+
+  if (typeof fn !== 'function') {
+    return false;
+  }
+
+  if (!name) {
+    return true;
+  }
+
+  if (!!fn._only && _.indexOf(fn._only, name) === -1) {
```

Comment:
It's better if we can create a map based on only and except routes
at the beginning. 
Then we don't need to use indexOf. 
Not only about a performance issue. It's easy to read without it.